app/addons/documents/doc-editor/components.react.jsx

```diff
@@ -218,44 +218,137 @@ var AttachmentsPanelButton = React.createClass({
     };
   },
 
+  getInitialState: function () {
+    return {
+      filter: '',
+      maxHeight: 'none'
+    };
+  },
+
+  componentDidMount: function () {
+    $(window).on('resize.attachment-filter', this.updateMaxHeight);
+    Stores.attachmentFilterStore.on('change', this.onChange);
+  },
+
+  componentWillUnmount: function () {
+    $(window).off('resize.attachment-filter', this.updateMaxHeight);
+    Stores.attachmentFilterStore.off('change', this.onChange);
+  },
+
+  onChange: function () {
+    this.setState({
+      maxHeight: this.state.maxHeight,
+      filter: Stores.attachmentFilterStore.filter()
+    });
+  },
+
+  getMaxHeight: function () {
+    var $btn = $(this._button);
```

Comment:
We React we shouldn't really have to ever use jQuery. I'm not sure why we need this.